### PR TITLE
[FLINK-15383][table sql / planner & legacy planner] Using sink schema field name instead of query schema field name for UpsertStreamTableSink.

### PR DESCRIPTION
**Pull request by leonardBang**
Using sink schema field name instead of query schema field name for UpsertStreamTableSink.

## What is the purpose of the change

*This pull request  fix sink schema field name was used in  UpsertStreamTableSink instead of query schema field name in blink planner and legacy planner.*

## Brief change log

  - *legacy planner: fix StreamPlanner and UpdatingPlanChecker*
  - *blink planner: fix StreamExecSink and UpdatingPlanChecker*

## Verifying this change
  - * add IT case in  TableSinkITCase for legacy planner and blink planner* 

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): ( no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by JingsongLi** [comment]
Is this for FLINK-15383?
"Using sink Schema field name instead of query Schema field name in UpsertStreamTableSink"

File: flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/table/TableSinkITCase.scala [hunk 1]
```diff
@@ -312,9 +312,9 @@ class TableSinkITCase extends AbstractTestBase {
 
     t.select('id, 'num, 'text.charLength() as 'len, ('id > 0) as 'cTrue)
       .groupBy('len, 'cTrue)
-      .select('len, 'id.count as 'cnt, 'cTrue)
-      .groupBy('cnt, 'cTrue)
-      .select('cnt, 'len.count as 'lencnt, 'cTrue)
+      .select('len, 'id.count as 'count, 'cTrue)
+      .groupBy('count, 'cTrue)
```
Comment (JingsongLi) [inline flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/table/TableSinkITCase.scala]: Add comment to show what are you test.

**Comment by leonardBang** [comment]
> Thanks @leonardBang , left some comments.

Thanks @JingsongLi, I'll update soon

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/util/UpdatingPlanChecker.scala [hunk 2]
```diff
@@ -37,8 +37,12 @@ object UpdatingPlanChecker {
   }
 
   /** Extracts the unique keys of the table produced by the plan. */
-  def getUniqueKeyFields(plan: RelNode): Option[Array[String]] = {
-    getUniqueKeyGroups(plan).map(_.map(_._1).toArray)
+  def getUniqueKeyFields(plan: RelNode, sinkFieldNames: List[String]): Option[Array[String]] = {
```
Comment (JingsongLi) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/util/UpdatingPlanChecker.scala]: Use array not use `List` too.

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/util/UpdatingPlanChecker.scala [hunk 3]
```diff
@@ -37,8 +37,12 @@ object UpdatingPlanChecker {
...
+    val relFieldNames = plan.getRowType.getFieldNames
+    getUniqueKeyGroups(plan).map(
+      _.map(
+        r => sinkFieldNames.get(relFieldNames.indexOf(r._1))
```
Comment (JingsongLi) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/util/UpdatingPlanChecker.scala]: Don't need new line.

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/UpdatingPlanChecker.scala [hunk 4]
```diff
@@ -20,7 +20,6 @@ package org.apache.flink.table.planner.plan.utils
 import org.apache.flink.table.planner.delegation.StreamPlanner
 import org.apache.flink.table.planner.plan.metadata.FlinkRelMetadataQuery
 import org.apache.flink.table.planner.plan.nodes.physical.stream._
-
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/UpdatingPlanChecker.scala]: Please keep this empty line.

File: flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/table/TableSinkITCase.scala [hunk 5]
```diff
@@ -312,9 +312,10 @@ class TableSinkITCase extends AbstractTestBase {
 
     t.select('id, 'num, 'text.charLength() as 'len, ('id > 0) as 'cTrue)
       .groupBy('len, 'cTrue)
-      .select('len, 'id.count as 'cnt, 'cTrue)
-      .groupBy('cnt, 'cTrue)
-      .select('cnt, 'len.count as 'lencnt, 'cTrue)
+      // test query field name is different with registered sink field name
+      .select('len, 'id.count as 'count, 'cTrue)
+      .groupBy('count, 'cTrue)
+      .select('count, 'len.count as 'lencnt, 'cTrue)
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/table/TableSinkITCase.scala]: This case doesn't reproduce the problem. You should also update `TestingUpsertTableSink#expectedKeys` to make sure `setKeyFields()` will verify the mapping between keys parameter and schema fields.

**Comment by leonardBang** [comment]
@wuchong Could you have a more look? thanks.

**Comment by wuchong** [comment]
Waiting for travis.

**Comment by JingsongLi** [comment]
LGTM thanks @wuchong and @leonardBang

**Comment by leonardBang** [comment]
@wuchong @JingsongLi Thanks very much.

**Pull request merged**